themes/default/source/js/application.js

```diff
@@ -472,6 +472,30 @@ function validateSize(map) {
   });
 }
 
+/**
+* Translates the X-position of an element inside a container so that its contents
+* are contained
+*
+* @param {object} element to position
+* @param {object} container element
+* @param {number} container margin
+*/
+function setPositionInContainer(el, container, margin = 0) {
+  const elRect = el.getBoundingClientRect();
+  const containerRect = container.getBoundingClientRect();
+
+  const leftDiff = elRect.left - containerRect.left;
+  const rightDiff = containerRect.right - elRect.right;
+  const halfElWidth = elRect.width/2;
```

Comment:
Adding a comment on `halfElWidth` would be kind to our future selves; we are not likely to remember that the element selector already translates the position to be centered above an item so this must be half its width.